osbuild/pipeline.py

```diff
@@ -330,16 +334,22 @@ def mark_pipeline(pl):
             if pl.assembler:
                 mark_assembler(pl.assembler)
 
-        for pl in self.pipelines:
+        for pl in self.pipelines.values():
             mark_pipeline(pl)
 
         return points
 
-    def __getitem__(self, pipeline_id):
-        for pl in self.pipelines:
-            if pl.id == pipeline_id:
+    def __getitem__(self, name_or_id: str) -> Pipeline:
+        pl = self.pipelines.get(name_or_id)
+        if pl:
+            return pl
+        for pl in self.pipelines.values():
+            if pl.id == name_or_id:
                 return pl
-        raise KeyError("{pipeline_id} not found")
+        raise KeyError("{name_or_id} not found")
```

Comment:
Missing f-string prefix

Reply:
Fixed in a later commit. I might have forgotten to mention that in the corresponding later commit though.

Reply:
oh, didn't notice the [Outdated], I usually look out for those.